File: src/main/java/io/creedengo/android/kotlin/AndroidKotlinRulesDefinition.java
/*
 * creedengo - Android Kotlin language - Provides rules to reduce the environmental footprint of your Kotlin Android applications
 * Copyright © 2026 Green Code Initiative (https://green-code-initiative.org/)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package io.creedengo.android.kotlin;

import java.util.ArrayList;
import java.util.List;

import org.sonar.api.SonarRuntime;
import org.sonar.api.server.rule.RulesDefinition;
import org.sonarsource.analyzer.commons.RuleMetadataLoader;

public class AndroidKotlinRulesDefinition implements RulesDefinition {

    static final String LANGUAGE = "kotlin";
    static final String REPOSITORY_KEY = "creedengo-android-kotlin";
    static final String REPOSITORY_NAME = "Creedengo Android Kotlin";
    static final String RESOURCE_BASE_PATH = "io/creedengo/android/kotlin/rules";

    private final SonarRuntime sonarRuntime;

    public AndroidKotlinRulesDefinition(SonarRuntime sonarRuntime) {
        this.sonarRuntime = sonarRuntime;
    }

    @Override
    public void define(Context context) {
        NewRepository repository = context.createRepository(REPOSITORY_KEY, LANGUAGE).setName(REPOSITORY_NAME);

        RuleMetadataLoader ruleMetadataLoader = new RuleMetadataLoader(RESOURCE_BASE_PATH, sonarRuntime);
        ruleMetadataLoader.addRulesByAnnotatedClass(repository, new ArrayList<>(AndroidKotlinCheckList.checks()));

        repository.done();
    }
}

File: src/main/kotlin/io/creedengo/android/kotlin/AndroidKotlinRulesDefinition.kt
/*
 * creedengo - Android Kotlin language - Provides rules to reduce the environmental footprint of your Kotlin Android applications
 * Copyright © 2026 Green Code Initiative (https://green-code-initiative.org/)
 *
 * Licensed under the GNU General Public License v3 — see LICENCE.md for details.
 */
package io.creedengo.android.kotlin

/**
 * Shared identifiers for the Creedengo Android Kotlin rule set.
 *
 * The plugin no longer declares its own [org.sonar.api.server.rule.RulesDefinition]:
 * sonar-kotlin owns repository creation via [CreedengoKotlinExtensionsProvider],
 * and metadata is auto-loaded from `org/sonar/l10n/kotlin/rules/kotlin/`.
 *
 * This object only exposes the constants still referenced elsewhere
 * (notably by [AndroidKotlinProfile]).
 */
object AndroidKotlinRulesDefinition {
    const val LANGUAGE = "kotlin"
    const val REPOSITORY_KEY = "creedengo-android-kotlin"
    const val REPOSITORY_NAME = "Creedengo Android Kotlin"
}
